src/transformers/models/seggpt/modeling_seggpt.py

```diff
@@ -0,0 +1,919 @@
+            torch.nn.init.normal_(module.type_token_semantic, std=std)
+            torch.nn.init.normal_(module.type_token_instance, std=std)
+
+
+SEGGPT_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`SegGPTConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+SEGGPT_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`SegGPTImageProcessor.__call__`]
+            for details.
+
+        prompt_pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Prompt pixel values. Prompt pixel values can be obtained using [`AutoImageProcessor`]. See
+            [`SegGPTImageProcessor.__call__`] for details.
+
+        prompt_masks (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Prompt mask. Prompt mask can be obtained using [`AutoImageProcessor`]. See [`SegGPTImageProcessor.__call__`] for
+            details.
+
+        bool_masked_pos (`torch.BoolTensor` of shape `(batch_size, num_patches)`, *optional*):
+            Boolean masked positions. Indicates which patches are masked (1) and which aren't (0).
+
+        embedding_type (`str`, *optional*):
+            Embedding type. Indicates whether the prompt is a semantic or instance embedding. Can be either
+            instance or semantic.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare SegGPT Model transformer outputting raw hidden-states without any specific head on top.",
+    SEGGPT_START_DOCSTRING,
+)
+class SegGPTModel(SegGPTPreTrainedModel):
+    def __init__(self, config: SegGPTConfig):
+        super().__init__(config)
+        self.config = config
+
+        self.embeddings = SegGPTEmbeddings(config)
+        self.encoder = SegGPTEncoder(config)
+        self.decoder = SegGPTDecoder(config)
```

Comment:
this needs to become part of the head models, `SegGPTForInstanceSegmentation` and `SegGPTForSemanticSegmentation`. The base `SegGPTModel` should only output the final hidden states similar to `BertModel` or `Gpt2Model`.

Reply:
Yeah not a big deal for me, Also we need `SegGptXXX` camel case is the default now everywhere now

Reply:
Changed to `SegGpt` https://github.com/huggingface/transformers/pull/27735/commits/449fa809882949d74c4e0c98f3090f2a64be6f3d and I'll keep the current structure for now

Reply:
I'm actually in favor of having separate `SegGptModel` and `SegGptForImageSegmentation` classes, adding the decoder to the latter. Current design looks a bit non-Transformers to me. Let's make it like ViT or BERT.